packages/@lwc/synthetic-shadow/src/faux-shadow/slot.ts

```diff
@@ -172,4 +271,171 @@ defineProperties(HTMLSlotElement.prototype, {
+            const defaultContentFrag = getDefaultContentFragment(this);
+            insertBefore.call(defaultContentFrag, newChild, refChild);
+            return newChild;
+        },
+    },
+    removeChild: {
+        writable: true,
+        enumerable: true,
+        configurable: true,
+        value<T extends Node>(this: HTMLSlotElement, oldChild: T): T {
+            const defaultContentFrag = getDefaultContentFragment(this);
+            removeChild.call(defaultContentFrag, oldChild);
+            return oldChild;
+        },
+    },
+    appendChild: {
+        writable: true,
+        enumerable: true,
+        configurable: true,
+        value<T extends Node>(this: HTMLSlotElement, newChild: T): T {
+            const defaultContentFrag = getDefaultContentFragment(this);
+            appendChild.call(defaultContentFrag, newChild);
+            return newChild;
+        },
+    },
+    replaceChild: {
+        writable: true,
+        enumerable: true,
+        configurable: true,
+        value<T extends Node>(this: HTMLSlotElement, newChild: Node, oldChild: T): T {
+            const defaultContentFrag = getDefaultContentFragment(this);
+            replaceChild.call(defaultContentFrag, newChild, oldChild);
+            return oldChild;
+        },
+    },
 });
+
+function hostInsertBefore(
+    slottedRecords: SlottedNodeRecord[],
+    slotElements: Record<string, HTMLSlotElement>,
+    nr: SlottedNodeRecord,
+    slotName: string,
+    refIndex: number
+) {
+    // insert right after ref
+    ArraySplice.call(slottedRecords, refIndex, 0, nr);
+    if (slotElements[slotName]) {
+        // needs immediate allocation
+        let refNode: Node | null = null;
+        // trying to find a good ref that was attached to the same slot
+        for (let i = refIndex + 1, len = slottedRecords.length; i < len; i += 1) {
+            if (slottedRecords[i][1] === slotName) {
+                refNode = slottedRecords[i][0];
+                break;
+            }
+        }
+        insertIntoSlot(slotElements[slotName], nr[0], refNode);
+    }
+}
+
+function hostRemoveChild(
+    slottedRecords: SlottedNodeRecord[],
+    slotElements: Record<string, HTMLSlotElement>,
+    slotName: string,
+    index: number
+) {
+    // remove record from records
+    const r = slottedRecords[index];
+    ArraySplice.call(slottedRecords, index, 1);
+    if (slotElements[slotName]) {
+        // needs immediate removal from dom
+        removeFromSlot(slotElements[slotName], r[0]);
+    }
+}
+
+/**
+ * This operation will add few methods to the host to attempt to control
+ * the insertion and removal of child nodes that must be slotted into its
+ * shadow root.
+ *
+ * Note: this method does not support innerHTML, and co., it only support
+ *       regular dom manipulation methods.
+ */
+export function adoptHostToSupportSlotting(host: Element) {
+    defineProperties(host, {
+        insertBefore: {
```

Comment:
we don't cover anything... we only cover insertBefore and removeChild, which are used by the engine diffing algo, plus appendChild and replaceChild since those are common in other scenarios... we do not support .innerHTML, textContent, nodeValue, etc... we don't support them today anyway.